cmd/software/deploy.go

```diff
@@ -51,6 +52,7 @@ func NewDeployCmd() *cobra.Command {
 	cmd.Flags().BoolVarP(&saveDeployConfig, "save", "s", false, "Save deployment in config for future deploys")
 	cmd.Flags().BoolVarP(&ignoreCacheDeploy, "no-cache", "", false, "Do not use cache when building container image")
 	cmd.Flags().StringVar(&workspaceID, "workspace-id", "", "workspace assigned to deployment")
+	cmd.Flags().BoolVarP(&isDagOnlyDeploy, "dag-only", "", false, "Deploy only those dags which are present in the /dags folder")
```

Comment:
Do we want to provide this feature flag for all software platform versions, if not we would need to shield it. Attaching examples for reference:
https://github.com/astronomer/astro-cli/blob/main/cmd/airflow.go#L154-L155
https://github.com/astronomer/astro-cli/blob/main/cmd/software/deployment.go#L223-L225

Reply:
Sure, let me add it 👍

Reply:
Well, what's the expectation when a user uses the new feature flag with an older Software version?
I mean it feels misleading to say to the user ``` to perform this operation, set both deployments.dagOnlyDeployment and deployments.configureDagDeployment to true in the Astronomer Platform``` when that configuration is not available for older Houston versions

Reply:
Updated the msg

Reply:
Added the software version check